### ISSUE TITLE
setup.py is out of sync with pyproject.toml

### ISSUE THREAD

**Issue by orionlee** [lightkurve/lightkurve#1181]
#### Problem description

[`setup.py`](https://github.com/lightkurve/lightkurve/blob/ad8eb62ae0f6b9d7ecf4183c391fe9891a62a969/setup.py) is now out of sync with the (authoritative) `pyproject.toml`, e.g., `setup.py` still states minimum astropy is 4.1

Th comments in `setup.py` suggests that it should be automatically generated by [dephell](https://github.com/dephell/dephell), but it is no longer actively maintained. However, when I tried to use the latest version to convert `pyproject.toml` to `setup.py`, I got a weird error (something about a required positional argument `trivia` is missing).

Issue:
- should we continue to maintain `setup.py`?
- if so, do we need something other than `dephell`?

#### Environment

-  lightkurve version (e.g. 1.0b6): 2.1.1

**Comment by barentsen** [comment]
Agreed.  I tried updating `setup.py` using `dephell` during the release of Lightkurve v2.1 and encountered the exact same problem, i.e., `dephell` is no longer maintained and raised an error I could not work around).

I believe we maintained `setup.py` only to enable lightkurve to be installed in editable mode (`pip install -e`), which was not previously possible using `pyproject.toml` and `poetry`.

Researching this again, it looks like `pip>=21.3` and `poetry-core>=1.0.8` now supports editable installs thanks to PEP 610.

It looks like this is working for me.  Using the older `pip==21.2.4` the editable install fails:

```
$ python -m venv create editabletest
$ source editabletest/bin/activate
$ mv ~/dev/lightkurve/setup.py /tmp/
$ python -m pip install -e ~/dev/lightkurve
ERROR: File "setup.py" or "setup.cfg" not found. Directory cannot be installed in editable mode: /Users/gb/dev/lightkurve
(A "pyproject.toml" file was found, but editable mode currently requires a setuptools-based build.)
WARNING: You are using pip version 21.2.4; however, version 22.0.4 is available.
You should consider upgrading via the '/Users/gb/tmp/editabletest/bin/python -m pip install --upgrade pip' command.
```

However upgrading `pip` made it work!
```
$ python -m pip install --upgrade pip
Successfully installed pip-22.0.4
$ python -m pip install -e ~/dev/lightkurve
Successfully built lightkurve
Installing collected packages: webencodings, pytz, fbpca, certifi, zipp, urllib3, typing-extensions, tqdm, tornado, threadpoolctl, soupsieve, six, PyYAML, pyparsing, pillow, numpy, memoization, MarkupSafe, kiwisolver, joblib, idna, future, fonttools, cycler, charset-normalizer, uncertainties, scipy, requests, python-dateutil, pyerfa, patsy, packaging, Jinja2, importlib-metadata, html5lib, beautifulsoup4, autograd, scikit-learn, pandas, oktopus, matplotlib, keyring, bokeh, astropy, pyvo, astroquery, lightkurve
$ python -c "import lightkurve; print(lightkurve.__path__)"
['/Users/gb/dev/lightkurve/src/lightkurve']
```

In conclusion:
* I think we can remove `setup.py` and the dependency on `dephell`.
* Update the docs to explain editable installs require `pip>=21.3` and a recent version of poetry (TBD).

Any objections?

**Comment by orionlee** [comment]
Sounds good to me. 

On a related note, the minimum requirement on the doc is also out of date.
https://docs.lightkurve.org/about/install.html#requirements

We could manually update them. But it seems to be something that is bound to be out of sync because someone forgot to update them sooner or later. Thoughts?

**Comment by barentsen** [comment]
> On a related note, the minimum requirement on the doc is also out of date.

Thanks for pointing that out!

I'm tempted to say we ought to remove specific version numbers from that page.  Instead, we can have a sentence+link which explains that the exact requirements can be found in the pyproject.toml file, which is pretty readable.

**Comment by gully** [comment]
Interesting! A `setup.py` is normally my go-to for establishing the development environment, but upon seeing the `lightkurve` recs, I see there are alternatives.  Good to know! Thanks.
